forem/forem
# Fixes #21392 "Fix the backend and frontend logic -2"

## What type of PR is this? (check all applicable)


## Description

This PR resolves a bug where applying a "sort by" filter (e.g., "Newest") on the search results would return unrelated articles. The issue is addressed by ensuring that the search query (`search_fields`) persists when sorting is applied, and the sorting logic is applied only to the filtered results.

Changes made:
1. Backend: Updated `Homepage::ArticlesQuery` to include filtering by `search_fields` (the search query).
2. Frontend: Modified the `fetchFeedItems` function in `Feed.jsx` to include `search_fields` and `sort_by` in the API requests.

## Related Tickets & Documents

- Closes #21392

## QA Instructions, Screenshots, Recordings

QA Instructions:
1. Perform a search using a term like `welcome`.
   - Verify that the results contain only articles with "welcome" in their titles.
2. Apply sorting options like "Newest" or "Oldest."
   - Confirm that the search results remain filtered while being sorted as per the selected option.
3. Check the browser's Network Tab to ensure API requests include the correct parameters (`search_fields` and `sort_by`).
   -API Example: GET /stories/feed?search_fields=welcome&sort_by=published_at&sort_direction=asc

## UI accessibility checklist
N/A - No UI changes were made.

## Added/updated tests?

- [x] No, and this is why: Manual testing instructions have been provided. Automated tests for backend and frontend are recommended but were not implemented in this PR due to time constraints.
